bin/.upgrades/versions/v1/upgradeStorage.js

```diff
@@ -0,0 +1,389 @@
+    mappings = mappingsResponse.body[index].mappings[collection];
+
+  // remove obsolete mapping properties
+  if (mappings.properties && mappings.properties._kuzzle_info) {
+    delete mappings.properties._kuzzle_info.properties.active;
+    delete mappings.properties._kuzzle_info.properties.deletedAt;
+  }
+
+  await target.indices.putMapping({
+    index: newIndex,
+    body: {
+      properties: mappings.properties,
+      dynamic: mappings.dynamic || false,
+      _meta: mappings._meta
+    }
+  });
+}
+
+async function createNewIndex (source, target, index, collection, newIndex) {
+  const exists = await target.indices.exists({ index: newIndex });
+
+  if (exists.body) {
+    await target.indices.delete({ index:newIndex });
+  }
+
+  await target.indices.create({ index: newIndex });
+}
+
+async function upgrade (source, target, index, collection, newIndex) {
+  await createNewIndex(source, target, index, collection, newIndex);
+  await upgradeMappings(source, target, index, collection, newIndex);
+  return await moveData(source, target, index, collection, newIndex);
+}
+
+async function upgradeInternalStorage (context, source, target) {
+  const
+    config = context.config.services.storageEngine.internalIndex,
+    index = `${INTERNAL_PREFIX}${config.name}`,
+    mapconfig = config.collections,
+    collections = {
+      validations: mapconfig.validations,
+      config: mapconfig.config,
+      roles: mapconfig.roles,
+      users: null,
+      profiles: mapconfig.profiles
+    };
+
+  for (const [collection, mappings] of Object.entries(collections)) {
+    const newIndex = getNewIndexName(index, collection);
+    let total;
+
+    if (mappings) {
+      await createNewIndex(source, target, index, collection, newIndex);
+      await target.indices.putMapping({ index: newIndex, body: mappings });
+      total = await moveData(source, target, index, collection, newIndex);
+    }
+    else {
+      total = await upgrade(source, target, index, collection, newIndex);
+    }
+
+    context.log.ok(`... migrated internal data: ${collection} (${total} documents)`);
+  }
+
+  // bootstrap document
+  await target.create({
+    index: `${index}.config`,
+    id: 'internalIndex.dataModelVersion',
+    body: { version: '2.0.0' }
+  });
+
+  await target.create({
+    index: `${index}.config`,
+    id: `${config.name}.done`,
+    body: { timestamp: Date.now() }
+  });
+}
+
+async function upgradePluginsStorage (context, source, target) {
+  const
+    { body } = await source.cat.indices({ format: 'json' }),
+    indexes = body.map(b => b.index).filter(n => n.startsWith('%plugin:'));
+
+  for (const index of indexes) {
+    const
+      plugin = index.split(':')[1],
+      newIndexBase = `%plugin-${plugin}${NAME_SEPARATOR}`,
+      mappings = await source.indices.getMapping({ index }),
+      collections = Object.keys(mappings.body[index].mappings);
+
+    for (const collection of collections) {
+      const newIndex = newIndexBase + collection;
```

Comment:
```suggestion
      const newIndex = `${newIndexBase}${collection}`;
```
I know concatenation was enough but IMHO interpolation improve readability

Reply:
This is one of the few examples where I personally find concatenation more readable. I guess this is a subjective matter.

Reply:
What if `newIndexBase` and `collection` are integers? :stuck_out_tongue:

Reply:
Yes, I didn't say it but it's nitpicking ;)